### ISSUE TITLE
Add support for building, running, and testing multi-platform targets

### ISSUE THREAD

**Issue by pepicrft** [tuist/tuist#5920]
### What problem or need do you have?

x

### Potential solution

x

### macOS version

x

### Tuist version

x

### Xcode version

x

**Comment by waltflanagan** [comment]
What are we thinking about when we say "support"  what would `tuist build` do when run for a multiplatform target? build against every supported platform sequentially? 

Similarly with `run` what does that mean with a multiplatform target?

**Comment by pepicrft** [comment]
> What are we thinking about when we say "support" what would tuist build do when run for a multiplatform target? build against every supported platform sequentially?

Exactly. Also make sure that there's a CLI interface to indicate the platforms they want to build for.
Looking at the current interface, I think it's a bit confusing:

```
  -d, --device <device>   Build on a specific device.
  --platform <platform>   Build for a specific platform.
  -o, --os <os>           Build with a specific version of the OS.
```

Should we use `--destination` instead, like Xcode does? Instead of abstracting that away and trying to be smart about it?

> Similarly with run what does that mean with a multiplatform target?

I don't think this one makes sense to make it multi-platform. I can't think of a use case where the user will want to run it in three destinations at the same time? Or maybe it's common?

**Comment by woin2ee** [comment]
Hi! @pepicrft 👋
It's been a few months, but are you right in saying this because there are platform settings and device and OS settings separate each?

> Looking at the current interface, I think it's a bit confusing:
> 
> ```
>   -d, --device <device>   Build on a specific device.
>   --platform <platform>   Build for a specific platform.
>   -o, --os <os>           Build with a specific version of the OS.
> ```

If yes, I think we need integrated platform settings because device and OS settings are dependent on a platform. What do you think now? still have the same think?

> Should we use `--destination` instead, like Xcode does? Instead of abstracting that away and trying to be smart about it?

**Comment by pepicrft** [comment]
> It's been a few months, but are you right in saying this because there are platform settings and device and OS settings separate each?

Correct. As a user of the CLI, when I look at that help menu, I don't really know which values I should be able to pass, which makes me think about whether we should just embrace the `-destination` interface of the underlying `xcodebuild` process.

```bash
tuist build -- -destination '....'
```

How much value are we providing by adding another interface on top of that (if any)?

**Comment by hanrw** [comment]
+1

```
 tuist test    
Generating project for testing
Loading and constructing the graph
It might take a while if the cache is empty
Using cache binaries for the following targets: 
Generating workspace SubscriptionCalendar.xcworkspace
Generating project SubscriptionCalendar
Resolving package dependencies using xcodebuild
Testing scheme SubscriptionCalendar-Workspace


✖ Error 
  Only single platform targets supported. The target DomainTests specifies multiple supported platforms (macos, ios). 
```

**Comment by krapcys1-maker** [comment]
Hi 👋
I’d like to work on this issue.
Is it still available?

**Comment by doovers** [comment]
/attempt #5920

Planning to fix the multi-platform target support for `tuist build`, `tuist test`, and `tuist run` commands by:
- Removing the error thrown when a target supports multiple platforms for build/test
- Iterating over all supported platforms sequentially when `--platform` is not specified
- Adding `--platform` flag to `tuist run` for cases where the target supports multiple platforms (since running on multiple platforms simultaneously doesn't make sense)

**Comment by Nithinfgs** [comment]
/attempt #5920

Implementation Plan

Add a new XcodeBuildDestination model and XcodeBuildPlatform enum to represent destinations.
Add --destination flag support to BuildCommand, TestCommand, and RunCommand.
Update BuildService and TestService to iterate through all platforms when a scheme is multi-platform and no destination is provided.
Require --destination in RunService when running multi-platform targets.
Add unit tests for destination parsing and build service behavior.

**Comment by devman247** [comment]
Hi, I am interested in solving this issue. Please send me an invitation to join.
